src/providers/data_provider/dp.c

```diff
@@ -199,10 +199,7 @@ dp_init_send(TALLOC_CTX *mem_ctx,
     ret = EAGAIN;
 
 done:
-    if (ret == EOK) {
-        tevent_req_done(req);
-        tevent_req_post(req, ev);
-    } else if (ret != EAGAIN) {
+    if (ret != EAGAIN) {
```

Comment:
Why?

Reply:
`ret` will be `ENOMEM` or `EAGAIN` at this point of execution, so it isn't possible for it to be `EOK`.

Reply:
Yes, it's really strange because it seems like the happy path is supposed to return `EAGAIN`. Just by changing `EAGAIN` to `EOK` it would make sense.

@pbrezina can you check it? I know it's been some time but you changed this part of the [code](https://github.com/SSSD/sssd/commit/c7e2d7a56d8969f26921811e5122f78b257fb51e).

Reply:
Function returns a request, not a code. Technically implementation is ok, but style is a bit weird.

Reply:
Hi,

`EAGAIN` is used in the tevent-request code to indicate to the caller that the request is still running and should stay in the event loop.

HTH

bye,
Sumit

Reply:
> Hi,
> 
> `EAGAIN` is used in the tevent-request code to indicate to the caller that the request is still running and should stay in the event loop.
> 
> HTH
> 
> bye, Sumit

Maybe 'caller' does not fit here well, better `EAGAIN` is used in general in the tevent-request code to indicate that the request is still running.

Reply:
As Sumit said, this is our tevent convention. I think it is not yet documented in our coding style guide.
* EOK -> finished, call `tevent_req_done`
* EAGAIN -> continue, asynchronous processing
* other -> error, call `tevent_req_error`